agent/ui_endpoint_test.go

```diff
@@ -1397,34 +1398,59 @@ func TestUIServiceTopology(t *testing.T) {
 		}
 	}
 
-	t.Run("request without kind", func(t *testing.T) {
-		req, _ := http.NewRequest("GET", "/v1/internal/ui/service-topology/ingress", nil)
-		resp := httptest.NewRecorder()
-		obj, err := a.srv.UIServiceTopology(resp, req)
-		require.Nil(t, err)
-		require.Nil(t, obj)
-		require.Equal(t, "Missing service kind", resp.Body.String())
-	})
-
-	t.Run("request with unsupported kind", func(t *testing.T) {
-		req, _ := http.NewRequest("GET", "/v1/internal/ui/service-topology/ingress?kind=not-a-kind", nil)
-		resp := httptest.NewRecorder()
-		obj, err := a.srv.UIServiceTopology(resp, req)
-		require.Nil(t, err)
-		require.Nil(t, obj)
-		require.Equal(t, `Unsupported service kind "not-a-kind"`, resp.Body.String())
-	})
+	type testCase struct {
```

Comment:
An alternative is to use an anonymous struct for these, since the type is only used once:

```suggestion
	tcs := []struct {
        name    string
		httpReq *http.Request
		want    *ServiceTopology
		wantErr string
    }{
        {
        	...
        },
    } 
```

Will leave that decision up to you though.

Reply:
With the pattern I mention below, it is required to have a named `type testCase struct` as the first section of the table test.